exercises/security-rnd/software-engineer/README.md

```diff
@@ -20,20 +20,23 @@ You can obtain package data through the npm registry: `https://registry.npmjs.or
 
 #### Things to consider
 
-  1. Look at the inner "dependencies" object for analysis of first-order dependencies.
+  1. Look at the inner "dependencies" object for analysis of its dependencies and do so recursively.
   2. There are currently over 800K packages on npmjs.com, and the number is growing all the time.
   3. The packages update from time to time, just as their dependencies.
   4. What makes a good web service? API, architecture, data storage, low latency, scalability, monitoring, you name it :)
   5. Consider the quality and structure of your codebase; is it maintainable?
-  6. Consider production readiness (to some extent) and is it safe to deploy changes?
+  6. Consider production readiness (to some extent) 
+ 
 
-#### Implementation
+#### Implementation 
 
-  1. A working web application that, given a name of an npm published package, returns the set of dependencies for said package.
-  2. Present the dependencies in a tree view.
-  3. Account of asyncronous fetching of dependencies as you see fit.
-  4. Cache relevant data so that repeated requests resolve with minimum latency.
+  1. A working web application that, given a name of an npm published package, returns the set of the (recursive) dependencies for said package. Consider that different types of clients can make use of this endpoint.
+  2. The endpoint should return the recursive dependencies of a package as a dependency tree - should you get the whole dependency tree all at once or not? Why? Please justify this in your README
+  3. Account of asynchronous fetching of dependencies as you see fit.
```

Comment:
I think we should remove this. As I explained in the google doc, I think this is just an artifact from copying the exercise over from the node/typescript version.

I have a feeling its confusing a lot of our candidates especially because we strongly suggest them to use python as the language for implementation.

Reply:
Fair, but when implementing the solution to get the whole dependency tree, this should be async; otherwise each request will only happen after all the other previous requests have finished; although this might be also something that can be improved in the second round of the interview?

Because now we're being explicit about the fact that we need the whole dependency tree in the response

Reply:
Yes, please remove the reference to asynchronous. 

What matters, and could be a good discussion point for the interview, are performances. If the application works well it could be because caching / multi processing / async (or all of them :D )

Reply:
> Fair, but when implementing the solution to get the whole dependency tree, this should be async; otherwise each request will only happen after all the other previous requests have finished;

True, but there could be other ways round that. In the end, serially requesting the data is a perfectly reasonable approach if you can justify it with the correct cache setup. 

My exercise worked this way for example. I justified it because I reckoned the cost for introducing async python (especially back when it was still very new) was not worth it when a cache layer which was warmed up with handle the vast majority of cases.

We should basically leave it up to the candidate to decide, justify and explain their reasoning without pushing them to a specific technology

Reply:
```suggestion
  3. Account the performance of fetching the dependencies as you see fit. 
```